### PR TITLE
WIP: Update to 0.4.1

### PR DESCRIPTION
**Pull request by thomasjpfan**
This PR updates skorch to support PyTorch 0.4.1. There is one small optimization to `skorch.utils. to_tensor` by using [`torch.as_tensor`](https://pytorch.org/docs/stable/torch.html#torch.as_tensor), which never copies unless necessary.

### CODE REVIEW

**Comment by taketwo** [comment]
Maybe also update import/docs for `Subset` since it was moved (https://github.com/pytorch/pytorch/pull/7816).

**Comment by BenjaminBossan** [comment]
Nice and quick work. I will have a deeper look at this on Monday. Too bad that this came shortly after our release ;) Are the changes breaking or only deprecated?

**Comment by taketwo** [comment]
Subset is not breaking.

>  Too bad that this came shortly after our release ;)

What about skorch 0.3.1? :)

**Comment by thomasjpfan** [comment]
There aren't any breaking changes:

1. `torch.nn.functional.sigmoid` was deprecated in favor of  `torch.sigmoid`
2. `torch.nn.functional.tanh` was deprecated in favor of  `torch.tanh`
3. `torch.utils.data.dataset.Subset` can now be imported up a level, `torch.utils.data.Subset`.

**Comment by thomasjpfan** [comment]
Since the PyTorch version on pypi is built on cuda-9.0, this PR also updates the Dockerfile to use cuda-9.0.

**Comment by benjamin-work** [comment]
LGTM, @ottonemo do you see anything missing?
